File: Week 2/PSet/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

int Coleman(string sen);

int main(void)
{
    // Get string input from user
    string sentence = get_string("Text: ");
    // Grade is the return of the function
    int grade = Coleman(sentence);
    // if its lower than 1 its before grade 1
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    // if its 16 or higer its Grade 16+
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    // else its that grade
    else
    {
        printf("Grade %i\n", grade);
    }
}

int Coleman(string sen)
{
    // initializing variables outside the loop
    int letter = 0;
    int word = 0;
    int sentence = 0;

    // iterating through the string
    for (int i = 0; i < strlen(sen); i++)
    {
        // if its a-z or A-Z increment letter
        if ((64 < sen[i] && sen[i] < 91) || (96 < sen[i] && sen[i] < 123))
        {
            letter++;
        }
        // if its . or ! or ? increment sentence
        else if (sen[i] == 46 || sen[i] == 33 || sen[i] == 63)
        {
            sentence++;
        }
        // if its space increment the word count
        else if (sen[i] == 32)
        {
            word++;
        }
    }
    // calculating the index
    word++;
    float L = letter / (float) word * 100;
    float S = sentence / (float) word * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    return (int) round(index);
}

File: Week 6/sentimental-readability/readability.py
def Coleman(sen):
    # initializing variables outside the loop
    letter = 0
    word = 0
    sentence = 0

    # iterating through the string
    for i in range(len(sen)):
        # if its a-z or A-Z increment letter
        asci = ord(sen[i])
        if (64 < asci and asci < 91) or (96 < asci and asci < 123):
            letter += 1
        # if its . or ! or ? increment sentence
        elif asci == 46 or asci == 33 or asci == 63:
            sentence += 1
        # if its space increment the word count
        elif asci == 32:
            word += 1
    # calculating the index
    word += 1
    L = letter / word * 100;
    S = sentence / word * 100;
    index = 0.0588 * L - 0.296 * S - 15.8;

    return round(index)


# Get string input from user
sentence = input("Text: ")
# Grade is the return of the function
grade = Coleman(sentence)
# if its lower than 1 its before grade 1
if grade < 1:
    print("Before Grade 1")
# if its 16 or higer its Grade 16+
elif grade >= 16:
    print("Grade 16+")
# else its that grade
else:
    print(f"Grade {grade}")